semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -526,6 +526,36 @@ trait TextDocumentOps { self: SemanticdbOps =>
         synthetics = finalSynthetics
       )
     }
+    def toJavaTextDocument: s.TextDocument = {
+      val symbols = List.newBuilder[s.SymbolInformation]
+      object traverser extends g.Traverser {
+        override def traverse(tree: g.Tree): Unit = {
+          tree match {
+            case d: g.DefTree =>
+              val _ = d.symbol.info // complete symbol
```

Comment:
Why is this necessary though? Scalac initializes all def symbols in Scala files during typechecking. Does this mean that Java files are treated differently?

Reply:
I've added a comment to explain why this is necessary
```
              // Unlike for Scala compilation units, def symbols in Java compilation units
              // are not initialized during type-checking. Without an explicit call to
              // initialize, some Java def trees will not have their infos set.
              val _ = d.symbol.initialize
```

If you remove this line the tests fail.

Reply:
Good to know!